fail2ban/ipaddr.py

```diff
@@ -163,132 +190,108 @@ def __radd__(self, other):
 		return "%s%s" % (other, self)
 
 	def __hash__(self):
-		return hash(self.addr)^hash((self.plen<<16)|self.family)
+		return hash(self._addr) ^ hash((self._plen << 16) | self._family)
 
+	@property
 	def hexdump(self):
-		""" dump the ip address in as a hex sequence in
-			network byte order - for debug purpose
+		"""Hex representation of the IP address (for debug purposes)
 		"""
 		if self.family == socket.AF_INET:
-			return "%08x" % self.addr
+			return "%08x" % self._addr
 		elif self.family == socket.AF_INET6:
-			return "%032x" % self.addr
+			return "%032x" % self._addr
 		else:
 			return ""
-	
+
+	# TODO: could be lazily evaluated
+	@property
 	def ntoa(self):
-		""" represent IP object as text like the depricated 
-			C pendant inet_ntoa() but address family independent
+		""" represent IP object as text like the deprecated
+			C pendant inet.ntoa but address family independent
 		"""
-		if self.family == socket.AF_INET:
+		if self.isIPv4:
 			# convert network to host byte order
-			binary = struct.pack("!L", self.addr)
-		elif self.family == socket.AF_INET6:
+			binary = struct.pack("!L", self._addr)
+		elif self.isIPv6:
 			# convert network to host byte order
-			hi = self.addr >> 64
-			lo = self.addr & 0xFFFFFFFFFFFFFFFFL
+			hi = self._addr >> 64
+			lo = self._addr & 0xFFFFFFFFFFFFFFFFL
 			binary = struct.pack("!QQ", hi, lo)
 		else:
-			return self.getRaw()
+			return self._raw
 
-		return socket.inet_ntop(self.family, binary)
+		return socket.inet_ntop(self._family, binary)
 
 	def getPTR(self, suffix=""):
-		""" generates the DNS PTR string of the provided IP address object
-			if "suffix" is provided it will be appended as the second and top
+		""" return the DNS PTR string of the provided IP address object
+
+			If "suffix" is provided it will be appended as the second and top
 			level reverse domain.
-			if omitted it is implicitely set to the second and top level reverse
+			If omitted it is implicitly set to the second and top level reverse
 			domain of the according IP address family
 		"""
-		if self.family == socket.AF_INET:
-			reversed_ip = ".".join(reversed(self.ntoa().split(".")))
+		if self.isIPv4:
+			reversed_ip = self.ntoa.split(".")
```

Comment:
Yeah... Exploded ;-)

On April 24, 2016 9:47:31 AM EDT, "Alexander Köppe" <notifications@github.com> wrote:
>>  			domain of the according IP address family
>>  		"""
>> -		if self.family == socket.AF_INET:
>> -			reversed_ip = ".".join(reversed(self.ntoa().split(".")))
>> +		if self.isIPv4:
>> +			reversed_ip = self.ntoa.split(".")
>
>the name `reversed_ip` is somewhat misleading as it's not yet reversed.
>Maybe `exploded_ip`? Or keeping the `reversed()` call here. What do you
>think?
>
>---
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/koeppea/fail2ban/pull/2/files/b9adbadb970043133f13448b50780d6de4ab5f37#r60844855